### PR TITLE
Bug 1822720: Configure TLS for OVN metrics endpoints

### PR DESCRIPTION
**Pull request by dcbw**
Continuation of https://github.com/openshift/cluster-network-operator/pull/581 with a small bugfix to pass unit tests.

@juanluisvaladas

### CODE REVIEW

File: bindata/network/ovn-kubernetes/monitor.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-
+---
```
Comment (abhat) [inline bindata/network/ovn-kubernetes/monitor.yaml:1]: nit: the separator seems un-necessary since there is nothing before.

Comment (dcbw) [reply bindata/network/ovn-kubernetes/monitor.yaml:1]: @abhat yeah that part was in Juan's original PR

**Comment by dcbw** [comment]
OK, this is going to be tricky. We need the secret created by the service-ca-controller to bring up the kube-rbac-proxy to secure our metrics. But the service-ca-controller is a pod-network process, which means that the ovnkube pods have to be up. But they won't be up until the kube-rbac-proxy is running, which requires certificates, which requires service-ca-controller.

**Comment by danwinship** [comment]
> OK, this is going to be tricky. We need the secret created by the service-ca-controller to bring up the kube-rbac-proxy to secure our metrics. But the service-ca-controller is a pod-network process

Right. This is why we roll our own certificates for the OVN databases.

IIRC even if we made service-ca-controller hostNetwork it still depends on other things that aren't ready as early as we need it.

**Comment by juanluisvaladas** [comment]
/retest 
All look like flakes

**Comment by juanluisvaladas** [comment]
/close 
https://github.com/openshift/cluster-network-operator/pull/581 merged

**Pull request closed without merging**